### DISCUSSION TITLE
Actual visibility

### DISCUSSION THREAD

**Discussion by rasmussl** [LemonAppDev/konsist#1584] [Request a Feature]
### Tell us more.

Hi :)

First of all, thanks for an amazing tool!

Feature request:
It would be nice if we could identify the overall and actual visibility/accessibility of this function.

Problem:
If I have an internal or private class, the KoClassDeclaration for this class will have either `hasInternalModifier=true` or `hasPrivateModifier=true`. If there is a function within this class that doesn't have a modifier, because it's covered by the class' internal or private modifier, the KoFunctionDeclaration for this function with have `hasInternalModifier=false` or `hasPrivateModifier=false`.

Here it would be nice to have the possibility of doing something like `.isPublic()` that took parent modifier and visibility into account.

**Comment by igorwojda** [comment]
This is an interesting feature. We will take a look.

In the mean time you can use `containingDeclaration`:

```kotlin
Konsist
.scopeFromProject()
.classes()
.functions()
	.assertTrue {
    	val visibilityModifierProvider = it.containingDeclaration as KoVisibilityModifierProvider

	    // visibilityModifierProvider.hasPrivateModifier
    	// visibilityModifierProvider.hasInternalModifier
	    // ..
	}
```

**Reply by rasmussl** [reply]
Thanks :)
I'll try this.
The best thing I could come up with was this `it.isTopLevel && !it.hasPublicAndDefaultModifier`.
